File: LinearSearch.java
import java.util.Scanner; 

/**
 * LinearSearch Implementation (renamed from Insertion.java constraints context)
 * Time Complexity: Best: O(1) | Average: O(n) | Worst: O(n)
 * Space Complexity: O(1)
 */
public class LinearSearch { 
    
    public static void main(String[] args) { 
        Scanner sc = new Scanner(System.in); 
        int[] arr = new int[5]; 
        
        System.out.println("Enter 5 array elements: "); 
        for (int i = 0; i < arr.length; i++) { 
            arr[i] = sc.nextInt(); 
        } 
        
        System.out.println("\nArray elements:"); 
        for (int i = 0; i < arr.length; i++) { 
            System.out.print(arr[i] + " "); 
        } 
        System.out.println();
        
        System.out.println("\nLinear Search for target: 20"); 
        linearSearch(arr, 20); 
        
        sc.close();
    } 
    
    public static void linearSearch(int[] arr, int target) { 
        boolean isFound = false; 
        
        for (int i = 0; i < arr.length; i++) { 
            if (arr[i] == target) { 
                System.out.println("Element found at index: " + i + " with value: " + arr[i]); 
                isFound = true; 
                break; 
            } 
        } 
        
        if (!isFound) {
            System.out.println("Element not found in list!"); 
        }
    } 
} 

File: linear_search.cpp
#include <iostream>
#include <vector>

using namespace std;

void linearSearch(const vector<int>& arr, int target) {
    bool isFound = false;

    // Check each element sequentially until target is found.
    for (int i = 0; i < static_cast<int>(arr.size()); i++) {
        if (arr[i] == target) {
            cout << "Element found at index: " << i << " with value: " << arr[i] << "\n";
            isFound = true;
            break;
        }
    }

    if (!isFound) {
        // This runs only if the loop never found target.
        cout << "Element not found in list!\n";
    }
}

int main() {
    vector<int> arr(5);
    cout << "Enter 5 array elements:\n";
    for (int i = 0; i < static_cast<int>(arr.size()); i++) {
        cin >> arr[i];
    }

    cout << "\nArray elements:\n";
    for (int value : arr) {
        cout << value << " ";
    }
    cout << "\n";

    cout << "\nLinear Search for target: 20\n";
    // Search for a fixed target value from the original assignment.
    linearSearch(arr, 20);

    return 0;
}
